### ISSUE TITLE
Traefik 2.0 doesn't respect scheme https unless Kubernetes service port name starts with https

### ISSUE THREAD

**Issue by zyp** [containous/traefik#5044]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?
I tried to set up an IngressRoute for kubernetes-dashboard:
``` yaml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  namespace: kube-system
  name: kubernetes-dashboard
spec:
  entryPoints:
    - https
  routes:
  - match: Host(`kubernetes-dashboard.mydomain`)
    kind: Rule
    services:
    - name: kubernetes-dashboard
      port: 443
      scheme: https
  tls: {}
```

### What did you expect to see?
I expected Traefik to use https to connect to kubernetes-dashboard.

### What did you see instead?
Traefik tries to contact kubernetes-dashboard over http, gets rejected and returns an error 500.

`/api/rawdata` returns the following information about the service:
``` json
  "services": {
    "kube-system/kubernetes-dashboard-1ae763fd03106a337232@kubernetescrd": {
      "loadbalancer": {
        "servers": [
          {
            "url": "http://10.40.0.6:8443"
          }
        ],
        "passHostHeader": true
      },
      "usedBy": [
        "kube-system/kubernetes-dashboard-1ae763fd03106a337232@kubernetescrd"
      ],
      "serverStatus": {
        "http://10.40.0.6:8443": "UP"
      }
    }
  }
```

If I edit the service definition for kubernetes-dashboard to add a name for the port starting with https, it works correctly.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.0-alpha7
Codename:     faisselle
Go version:   go1.12.6
Built:        2019-06-21T15:59:46Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

``` yaml
spec:
  replicas: 1
  selector:
    matchLabels:
      app: traefik
  template:
    metadata:
      labels:
        app: traefik
    spec:
      serviceAccountName: traefik-ingress-controller
      containers:
        - name: traefik
          image: traefik:v2.0
          args:
            - --api
            - --accesslog
            - --entrypoints.http.Address=:80
            - --entrypoints.https.Address=:443
            - --providers.kubernetescrd
            - --serverstransport.insecureskipverify
            - --acme
            - --acme.acmelogging
            - --acme.tlschallenge
            - --acme.onhostrule
            - --acme.email=REDACTED
            - --acme.entrypoint=https
            - --acme.storage=acme.json
          ports:
            - name: http
              containerPort: 80
            - name: https
              containerPort: 443
            - name: dashboard
              containerPort: 8080
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
time="2019-06-30T10:35:29Z" level=debug msg="Creating middleware" routerName=kube-system/kubernetes-dashboard-1ae763fd03106a337232@kubernetescrd serviceName=kube-system/kubernetes-dashboard-1ae763fd03106a337232 middlewareName=pipelining middlewareType=Pipelining entryPointName=https
time="2019-06-30T10:35:29Z" level=debug msg="Creating load-balancer" serviceName=kube-system/kubernetes-dashboard-1ae763fd03106a337232 entryPointName=https routerName=kube-system/kubernetes-dashboard-1ae763fd03106a337232@kubernetescrd
time="2019-06-30T10:35:29Z" level=debug msg="Creating server 0 http://10.40.0.6:8443" entryPointName=https routerName=kube-system/kubernetes-dashboard-1ae763fd03106a337232@kubernetescrd serviceName=kube-system/kubernetes-dashboard-1ae763fd03106a337232 serverName=0
time="2019-06-30T10:35:29Z" level=debug msg="Added outgoing tracing middleware kube-system/kubernetes-dashboard-1ae763fd03106a337232" middlewareType=TracingForwarder entryPointName=https routerName=kube-system/kubernetes-dashboard-1ae763fd03106a337232@kubernetescrd middlewareName=tracing
```

**Comment by jbdoumenjou** [comment]
Hi @zyp, thanks for your interest in the project.

The `scheme` option doesn't exist on service.
If we read the rawdata content, your server use the port 8443. 
To have a valid configuration, you need to use `https` as port name **OR** `443` as port.

Could you please try this and confirm if it works ?

**Comment by zyp** [comment]
As far as I can tell, the `scheme` option not only exists, but is *also* required. I tried multiple combinations; having a port named `https` but not setting the `scheme` option fails in the same way. Getting both the port name and the `scheme` option correct results in a https request.

My guess is that the original intention was to be able to both autodetect from the port name and override it with the `scheme` option, but then the logic got mixed up a bit and turned into an AND situation instead.

**Comment by ldez** [comment]
the `scheme` attribute doesn't exist in our [CRD](https://docs.traefik.io/v2.0/providers/kubernetes-crd/#traefik-ingressroute-definition)

https://github.com/containous/traefik/blob/06df6017dfc4464b81106e22bd7fcc61de5c3786/pkg/provider/kubernetes/crd/traefik/v1alpha1/ingressroute.go#L45-L50

**Comment by traefiker** [comment]
Closed by #5062.
